src/main/java/seedu/teachstack/logic/commands/DeleteCommand.java

```diff
@@ -36,17 +33,12 @@ public DeleteCommand(StudentId targetId) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Person> lastShownList = model.getAddressBook().getPersonList();
-
-        Optional<Person> personOptional = lastShownList.stream()
-                .filter(p -> p.getStudentId().equals(targetId))
-                .findFirst();
+        Person personToDelete = model.getPerson(targetId);
```

Comment:
This (and similar changes) look like a more concise way of obtaining the relevant person. Nicely done!